### PR TITLE
ISPN-11457 marshalling docs

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-11457 

needs backport to 10.1.x

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/proc_using_protostream_marshaller.adoc [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[id='protostream_cm_config']
+= Using the ProtoStream Marshaller
+{brandname} directly integrates a ProtoStream API to encode and decode Java
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_using_protostream_marshaller.adoc]: "integrates with the ProtoStream ..."

File: documentation/src/main/asciidoc/stories/assembly_marshalling.adoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+include::{topics}/proc_using_protostream_marshaller.adoc[leveloffset=+1]
+include::{topics}/proc_using_kyro_marshaller.adoc[leveloffset=+1]
+include::{topics}/proc_using_protostuff_marshaller.adoc[leveloffset=+1]
+include::{topics}/proc_using_custom_marshallers.adoc[leveloffset=+1]
+include::{topics}/proc_using_java_serialization.adoc[leveloffset=+1]
+include::{topics}/proc_using_jboss_marshaller.adoc[leveloffset=+1]
+include::{topics}/proc_adding_deserialization_whitelist.adoc[leveloffset=+1]
+include::{topics}/proc_storing_deserialized_objects.adoc[leveloffset=+1]
+include::{topics}/proc_enabling_store_binary.adoc[leveloffset=+1]
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/stories/assembly_marshalling.adoc]: At this point the Kryo and Protostuff marshallers are probably the marshallers with smallest amount of testing and usage in the wild, so we probably want to display the other marshaller implementations first.

File: documentation/src/main/asciidoc/topics/proc_using_kyro_marshaller.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
++
+. Specify the full path of implementing classes in your deployment JAR file within:
++
+----
+META-INF/services/org/infinispan/marshaller/kryo/SerializerRegistryService
+----
+
+//Community only
+ifndef::productized[]
+{brandname} provides a link:http://central.maven.org/maven2/org/infinispan/infinispan-marshaller-kryo-bundle/{infinispanversion}/infinispan-marshaller-kryo-bundle-{infinispanversion}.jar[Kryo Bundle] that includes all runtime class files for the Kryo marshalling implementation. Download the JAR and add it to the `server/lib` directory of your {brandname} servers.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_using_kyro_marshaller.adoc]: This only applies to server mode. It's possible to utilise the Kryo marshaller in Embedded mode by just adding the `infinispan-marshaller-kryo` jar as a dependency and then specifying the `org.infinispan.marshaller.kryo.KryoMarshaller` class as the marshaller.

File: documentation/src/main/asciidoc/topics/proc_using_protostuff_marshaller.adoc [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+You should then specify the full path of implementing classes in your deployment JAR file within:
++
+----
+META-INF/services/org/infinispan/marshaller/protostuff/SchemaRegistryService
+----
+
+//Community only
+ifndef::productized[]
+{brandname} provides a link:http://central.maven.org/maven2/org/infinispan/infinispan-marshaller-protostuff-bundle/{infinispanversion}/infinispan-marshaller-protostuff-bundle-{infinispanversion}.jar[Protostuff Bundle JAR] that includes all runtime class files for the Protostuff marshalling implementation. Download the JAR and add it to the `server/lib` directory of your {brandname} servers.
+endif::productized[]
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_using_protostuff_marshaller.adoc]: Same as Kryo, this doesn't cover the Embedded use-case.

File: documentation/src/main/asciidoc/topics/proc_using_jboss_marshaller.adoc [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+default marshaller in previous {brandname} versions.
+
+[NOTE]
+====
+* You should use serialization-based marshalling with {brandname} but use
+Protostream instead, which is a high-performance binary wire format that
+ensures backwards compatibility.
+
+* JBoss Marshalling is deprecated in {brandname} and will be removed in a future
+release. Along with JBoss Marshalling {brandname} deprecates the
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_using_jboss_marshaller.adoc]: " Along with JBoss Marshalling {brandname}" redundant? Can we just state that "JBoss Marshalling and the `AdvancedExternalizer` interface is deprecated in ..."

File: documentation/src/main/asciidoc/topics/proc_enabling_store_binary.adoc [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+The `equals()` method of the wrapper class either compares binary
+representations (byte arrays) or delegates to the wrapped object instance's
+`equals()` method, depending on whether both instances being compared are in
+serialized or deserialized form at the time of comparison. If one of the
+instances being compared is in one form and the other in another form, then one
+instance is either serialized or deserialized.
+
+This affects how keys stored in the cache work because comparisons happen on
+the key that is wrapped by a `MarshalledValue`.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_enabling_store_binary.adoc]: This is wrong. `MarshalledValue` is only ever used by stores and is not used to store entries in the cache.

File: documentation/src/main/asciidoc/topics/proc_implementing_serial_cxt_initializer.adoc [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+----
++
+[source,java]
+.AuthorMarshaller.java
+----
+include::code_examples/AuthorMarshaller.java[]
+----
++
+. Create a `SerializationContextInitializer` implementation that registers the
+`.proto` schema and marshallers with a `SerializationContext`.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_implementing_serial_cxt_initializer.adoc]: Maybe make it clear that we're talking about ProtoStream Marshallers here. Something like:

"schema and the ProtoStream marshaller implementations".

File: documentation/src/main/asciidoc/topics/proc_enabling_store_binary.adoc (hunk 6 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/proc_enabling_store_binary.adoc]: I just removed this line. tbh there were a few other things about store as binary in the doc that seemed off. I dropped some of the deprecated content. hopefully it's ok. if you don't mind taking another look I'd feel better about it. cheers.

File: documentation/src/main/asciidoc/topics/proc_enabling_store_binary.adoc [hunk 8]
```diff
@@ -0,0 +1,43 @@
+[id='store_as_binary']
+= Storing Data in Binary Format
+{brandname} can store data in its serialized form, in binary format, and then
+either serialize or deserialize Java objects as needed. This behavior is also
+referred to as lazy deserialization.
+
+When you store data in binary format, deserialization typically uses the thread
+context class loader of the invocation that requires deserialization, which
+provides an effective way to isolate classloaders.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/proc_enabling_store_binary.adoc]: I'm not sure this paragraph adds any value.

**Comment by ryanemerson** [comment]
@oraNod Happy St Patrick's day!!!

**Pull request merged**